File: Unit02/Loops.java
public class Loops {

    public static void printChars(String a_string) {
        System.out.println(a_string); // print (a_string)
        int length = a_string.length(); // len (a_string)
        int index = 0; // index = 0
        while (index < length) {
            char c = a_string.charAt(index); // c = a_string [index]
            System.out.println(index + " " + c);
            index += 1;
        }
    }

    public static void printCharsFor(String a_string) {
        System.out.println(a_string);
        for (int index=0; index<a_string.length(); index+=1) {
            char c = a_string.charAt(index);
            System.out.println(c);
        }
        
    }
        
    public static void main(String[] args) {
        printChars("Skywalker");

        for (int c=0; c<11; c++) {
            System.out.println (c);
        }
        // int n = 15;
        // int sum = 0;
        // int count = 0;
        // while (count <= n) {
        //     System.out.println(count);
        //     sum += count;
        //     count += 1;
        // }
        // System.out.println(sum);
    }
    
}

File: Unit02/loops.py
# Alex Tedesco

def countdown (number):
    sum = 0
    while number >= 0:
        print (number)
        sum = sum + number
        number = number - 1
    return sum

def count_up (number):
    sum = 0
    count = 0
    while count <= 10:
        print (count)
        sum += count
        count += 1
    return sum

def print_chars_while_loop (a_string):
    index = 0
    length = len (a_string)
    while index < length:
        char = a_string [index]
        print (index, char)
        index += 1

def print_chars_for_loop (a_string):
    for each_char in a_string:
        print (each_char)

def main ():
    # print (countdown (10))
    # print (count_up (1))
    print (print_chars_while_loop ("Skywalker"))
    print (print_chars_for_loop ("Skywalker"))


if __name__ == "__main__":
    main ()